Tests/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ include_directories(${PROJECT_SOURCE_DIR}/Library)
 foreach(t ${TESTS})
   set(test_executable test_${t})
   add_executable(${test_executable} ${test_executable}.c)
-  target_link_libraries(${test_executable} H2LIB ${gcov})
+  target_link_libraries(${test_executable} H2LIB ${gcov} -lm)
```

Comment:
I  think you'll want  to wrap this in an `if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")`. Or write a `try_compile` to figure out if `-lm` is needed and available.